test/setup-clear-govm.sh

```diff
@@ -54,100 +59,112 @@ EOF
+        cat >/etc/systemd/system/kubelet.service.d/10-cri.conf <<EOF
 [Unit]
 After=$cri_daemon.service
 EOF
+        # flannel + CRI-O + Kata Containers needs a crio.conf change (https://clearlinux.org/documentation/clear-linux/tutorials/kubernetes):
+        #    If you are using CRI-O and flannel and you want to use Kata Containers, edit the /etc/crio/crio.conf file to add:
+        #    [crio.runtime]
+        #    manage_network_ns_lifecycle = true
+        #
+        # We don't use Kata Containers, so that particular change is not made to /etc/crio/crio.conf
+        # at this time.
+
+        # /opt/cni/bin is where runtimes like CRI-O expect CNI plugins. But cloud-native-basic installs into
+        # /usr/libexec/cni. Instructions at https://clearlinux.org/documentation/clear-linux/tutorials/kubernetes#id2
+        # are inconsistent at this time (https://github.com/clearlinux/clear-linux-documentation/issues/388).
+        #
+        # We solve this by creating the directory and symlinking all existing CNI plugins into it.
+        mkdir -p /opt/cni/bin
+        for i in /usr/libexec/cni/*;do
+            ln -s $i /opt/cni/bin/
+        done
+
+        # Reconfiguration done, start daemons. Starting kubelet must wait until kubeadm has created
+        # the necessary config files.
+        systemctl daemon-reload
+        systemctl restart $cri_daemon || (
+            systemctl status $cri_daemon || true
+            journalctl -xe || true
+            false
+        )
+        systemctl enable $cri_daemon kubelet
+    fi
+}
+
+install_bundles
 
-    # flannel + CRI-O + Kata Containers needs a crio.conf change (https://clearlinux.org/documentation/clear-linux/tutorials/kubernetes):
-    #    If you are using CRI-O and flannel and you want to use Kata Containers, edit the /etc/crio/crio.conf file to add:
-    #    [crio.runtime]
-    #    manage_network_ns_lifecycle = true
-    #
-    # We don't use Kata Containers, so that particular change is not made to /etc/crio/crio.conf
-    # at this time.
-
-    # /opt/cni/bin is where runtimes like CRI-O expect CNI plugins. But cloud-native-basic installs into
-    # /usr/libexec/cni. Instructions at https://clearlinux.org/documentation/clear-linux/tutorials/kubernetes#id2
-    # are inconsistent at this time (https://github.com/clearlinux/clear-linux-documentation/issues/388).
-    #
-    # We solve this by creating the directory and symlinking all existing CNI plugins into it.
-    mkdir -p /opt/cni/bin
-    for i in /usr/libexec/cni/*;do
-        ln -s $i /opt/cni/bin/
-    done
-
-    # Reconfiguration done, start daemons. Starting kubelet must wait until kubeadm has created
-    # the necessary config files.
-    systemctl daemon-reload
-    systemctl restart $cri_daemon || (
-        systemctl status $cri_daemon || true
-        journalctl -xe || true
-        false
-    )
-    systemctl enable $cri_daemon kubelet
+function init_region(){
```

Comment:
Is there a reason why this is a function and "init kubernetes" isn't? That seems inconsistent.

Just wondering, don't bother changing it.

Reply:
This inconsistency is the result of moving this part of code from `test/setup-kubernetes.sh`.

Reply:
This actually points towards a bigger conceptual issue that I missed before: `init_region` isn't specific to the underlying OS and thus shouldn't be in `setup-fedora-govm.sh` and `setup-clear-govm.sh`. It's not much code, but it's still some code duplication.

Can you come up with a solution that puts `init_region` into a single file?

I don't want to stop you from implementing this, but let me point out the solution that I came up with in the meantime: https://github.com/pohly/pmem-CSI/commit/38af431d2be36db0957d5199da6c92cffdbbb013

Your argument was that `make run_dm_tests` will allow a developer to test changes to the device manager quickly while working on the code. On my desktop, that takes over 4 minutes for each invocation. `make test_e2e TEST_E2E_FOCUS=gotests.*pmem-device-manager` achieves the same in 9 seconds once the VM was created (which only needs to be done once per day).

Do you still think that these changes are necessary?

Reply:
> `make test_e2e TEST_E2E_FOCUS=gotests.*pmem-device-manager` achieves the same in 9 seconds once the VM was created

As a developer now I should remember the new way how I to run the device manager tests, which is as good as running a new make target in my opinion.

If we do not kill the test VM created after running device manager tests assuming that it can be reusable or make it a bit flexible to choosing VM say `make test TEST_VM=worker1` then the current solution also shouldn't consume more time.

Reply:
> This actually points towards a bigger conceptual issue that I missed before: `init_region` isn't specific to the underlying OS and thus shouldn't be in `setup-fedora-govm.sh` and `setup-clear-govm.sh`. It's not much code, but it's still some code duplication.
> 

I myself had the same concern while doing this change, wether to duplicate code or move it a separate file or to post `create_vms()` in `start-kubernetes.sh` script. But chose to duplicate as it's not so big code and assuming that we might not need to touch this in future.

But I agree with you, should avoid code duplication.

> Can you come up with a solution that puts `init_region` into a single file?

Sure, I will work on this.

Reply:
> > Can you come up with a solution that puts `init_region` into a single file?
> 
> Sure, I will work on this.

Fixed this by moving the `init_region()`(renamed to `init_pmme_regions`)  to `start-kubernetes.sh` script and calling it after `create_vms()`.

Reply:
> make it a bit flexible to choosing VM say `make test TEST_VM=worker1` then the current solution also shouldn't consume more time.

Made these changes to `run_dm_test` target, so that one can choose a cluster where these tests should be run. It uses the `pmem-csi-${CLUSTER}-worker1` to run the tests. If this node already set up(say via `make start`) it just copies the binary and runs the tests.